src/renderer/features/network/CustomRpcForm/model/add-custom-rpc-model.ts

```diff
@@ -0,0 +1,179 @@
+import { createEffect, createEvent, createStore, sample } from 'effector';
+import { createForm } from 'effector-forms';
+
+import { networkService, RpcValidation } from '@shared/api/network';
+import { ExtendedChain } from '@entities/network';
+import { CustomRpcForm, NodeExistParam, RpcCheckResult } from '../lib/types';
+import { manageNetworkModel } from '@pages/Settings/Networks/model/manage-network-model';
+import { fieldRules } from '../lib/utils';
+
+const $addCustomRpcForm = createForm({
+  fields: {
+    name: {
+      init: '',
+      rules: fieldRules.name,
+    },
+    url: {
+      init: '',
+      rules: fieldRules.url,
+    },
+  },
+  validateOn: ['blur'],
+});
+
+const formInitiated = createEvent();
+const networkChanged = createEvent<ExtendedChain>();
+const rpcConnectivityChecked = createEvent<RpcCheckResult>();
+const nodeExistChecked = createEvent<boolean>();
+const processStarted = createEvent<boolean>();
+
+const $selectedNetwork = createStore<ExtendedChain | null>(null);
+const $rpcConnectivityResult = createStore<RpcCheckResult>(RpcCheckResult.INIT);
+const $isNodeExist = createStore<boolean>(false);
+const $isProcessStarted = createStore<boolean>(false);
+
+const checkRpcNodeFx = createEffect(
+  async ({ network, url }: { network: ExtendedChain | null; url: string }): Promise<RpcCheckResult | null> => {
+    if (!network) return null;
+
+    try {
+      rpcConnectivityChecked(RpcCheckResult.LOADING);
```

Comment:
Better to avoid use of events inside effect. better to cover this behaviour with `sample`

Reply:
I've managed to remove the others, but in this case I'm not sure what you'd recommend, since this is a temporary state, to show the loading button. I thought about creating something like
```ts
const connectivityCheckStarted = createEvent<boolean>()
const $isValid = createStore(false)
```

which is arguably nicer, and I'm happy to do, however this `connectivityCheckStarted` still needs to be called in the effect unless I miss something. I don't know a way to call an event with a `sample`, unless it's done in a `fn`?

Reply:
You can use effect as target in sample.

And for loading button you can use [pending property](https://effector.dev/en/api/effector/effect/#properties-pending) for `effect`

Reply:
with the pending, this makes total sense yes, thank you.